torch/_higher_order_ops/invoke_subgraph.py

```diff
@@ -203,21 +225,28 @@ def invoke_subgraph_func(ctx, subgraph, identifier, operands):
     with ctx.redispatch_to_next() as m:
         # TODO(anijain2305) - Long term, it might be a bit restrictive to ban
         # mutation/aliasing. Investigate if there is a way to support this.
-        # TODO(anijain2305) - Short term, improve compilation time by
-        # skipping this check if the identifier has been seen before.
-        pre_dispatch = hasattr(ctx, "mode") and ctx.mode.pre_dispatch
-        if _has_potential_branch_input_mutation(
-            subgraph, unwrapped_operands, pre_dispatch=pre_dispatch
-        ):
-            raise UnsupportedAliasMutationException(
-                "One of invoke_subgraph hop might be modifying the input!"
-            )
-        if _has_potential_branch_input_alias(
-            subgraph, unwrapped_operands, pre_dispatch=pre_dispatch
-        ):
-            raise UnsupportedAliasMutationException(
-                "One of invoke_subgraph hop might be aliasing the input!"
-            )
+        skip_check = False
+        invoke_subgraph_cache = get_invoke_subgraph_cache()
+        if invoke_subgraph_cache:
+            if invoke_subgraph_cache.get_functional_tensor_entry(identifier):
+                skip_check = True
+
+        if not skip_check:
+            pre_dispatch = hasattr(ctx, "mode") and ctx.mode.pre_dispatch
+            if _has_potential_branch_input_mutation(
```

Comment:
Should be moved to Dynamo. If there is a mutation, don't even create invoke_subgraph call.